src/processing/DataManipulator.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from tqdm import tqdm
+import matplotlib.pyplot as plt
+from sklearn.feature_selection import chi2
+from sklearn.model_selection import cross_val_score, train_test_split
+from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer, TfidfTransformer
+
+
+def breakdown_cat_distribution(df:pd.DataFrame, cat_columns:list):
+    """
+    Displays classes distribution of categorial classes in absolute and relative values.
+    - df: DataFrame to be explroed
+    - cat_columns: categorical columns
+    """
+    print(f'DatFrame columns breakdown\n' + 20 * '=')
+    for col in cat_columns:
+        print(f'\nColumn `{col}` distribution')
+        tmp = pd.concat([
+                df[col].value_counts(normalize=False, dropna=False).apply('{:,}'.format).rename('count'),
+                df[col].value_counts(normalize=True, dropna=False).apply('{:.1%}'.format).rename('percentage')
+            ], axis=1)
+        display(tmp)
+
+def get_most_correlated_tokens(features:np.array, labels:pd.Series, vectorizer:TfidfVectorizer, category_to_id:dict, top_n_tokens:int=3):
+    """
+    Returns DataFrame with most correlated tokens of each category present in `category_to_id`, methodology is Chi Squared.
+    - features: numpy array of text features (result of vectorizer)
+    - labels: pandas Seires with target labels factorized (numeric values)
+    - vectorizer: vectorizer used for featurizing text
+    - top_n_tokens: top most correlated tokens to be returned
+    - category_to_id: dictionary to translate labels to human readable format
+    """
+
+    df = pd.DataFrame()
+    
+    for category, cat_id in sorted(category_to_id.items()):
+        features_chi2 = chi2(features, labels == cat_id)
+        indices = np.argsort(features_chi2[0])
+        feature_names = np.array(vectorizer.get_feature_names())[indices]
+        max_ngram = max([len(f.split(' ')) for f in feature_names])
+
+        sr = pd.Series(dtype=object)        
+        
+        for i in range(1, max_ngram+1):
+            ngrams = [v for v in feature_names if len(v.split(' ')) == i]
+            sr[f'{i}-grams'] = ngrams[-top_n_tokens:]
+
+        df[category] = sr
+        
+    return df.T
+
+def predict_text(text, classifier, vectorizer):
+    """
+    Returns predicted label of `text` based on `classifier` and `vectorizer`.
+    """
+    return {'text': text, 'prediction': classifier.predict(vectorizer.transform([text]))[0]}
```

Comment:
Why would you return the input text?

Reply:
Only for easier readability on the notebook!